init/fluent_bit_init_process.go

```diff
@@ -0,0 +1,400 @@
+
+	var metadata ECSTaskMetadata
+
+	ecsTaskMetadataEndpointV4 := os.Getenv("ECS_CONTAINER_METADATA_URI_V4")
+	if ecsTaskMetadataEndpointV4 == "" {
+		logrus.Warnln("[FluentBit Init Process] Unable to get ECS Metadata, ignore this warning if not running on ECS")
+		return metadata
+	}
+
+	res, err := httpClient.Get(ecsTaskMetadataEndpointV4 + "/task")
+	if err != nil {
+		logrus.Fatalf("[FluentBit Init Process] Failed to get ECS Metadata via HTTP Get: %s\n", err)
+	}
+
+	response, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		logrus.Fatalf("[FluentBit Init Process] Failed to read HTTP response: %s\n", err)
+	}
+	res.Body.Close()
+
+	err = json.Unmarshal(response, &metadata)
+	if err != nil {
+		logrus.Fatalf("[FluentBit Init Process] Failed to unmarshal ECS metadata: %s\n", err)
+	}
+
+	arn, err := arn.Parse(metadata.ECS_TASK_ARN)
+	if err != nil {
+		logrus.Fatalf("[FluentBit Init Process] Failed to parse ECS TaskARN: %s\n", err)
+	}
+
+	resourceID := strings.Split(arn.Resource, "/")
+	taskID := resourceID[len(resourceID)-1]
+	metadata.ECS_TASK_ID = taskID
+	metadata.AWS_REGION = arn.Region
+	metadata.ECS_TASK_DEFINITION = metadata.ECS_FAMILY + ":" + metadata.ECS_REVISION
+
+	return metadata
+}
+
+// set ECS Task Metadata as environment variables in the fluent-bit-invoker.sh
+func setECSTaskMetadata(metadata ECSTaskMetadata, filePath string) {
+
+	t := reflect.TypeOf(metadata)
+	v := reflect.ValueOf(metadata)
+
+	invokerFile := openFile(filePath)
+	defer invokerFile.Close()
+
+	for i := 0; i < t.NumField(); i++ {
+		if v.Field(i).Interface().(string) == "" {
+			continue
+		}
+		writeContent := "export " + t.Field(i).Name + "=" + v.Field(i).Interface().(string) + "\n"
+		_, err := invokerFile.WriteString(writeContent)
+		if err != nil {
+			logrus.Errorln(err)
+			logrus.Fatalf("[FluentBit Init Process] Cannot write %s in the fluent-bit-invoker.sh\n", writeContent[:len(writeContent)-2])
+		}
+	}
+}
+
+// create Fluent Bit command to use "-c" to specify the new main config file
+func createCommand(command *string, filePath string) {
+	*command = *command + " -c " + filePath
+}
+
+// create a folder to store S3 config files user specified (download them from S3)
+func createS3ConfigFileFolder(folderPath string) {
+	os.Mkdir(folderPath, 0777)
+}
+
+// get our built in config files or files from s3
+// process built-in config files directly
+// add S3 config files to folder "/init/fluent-bit-init-s3-files/"
+func getAllConfigFiles() {
+
+	// get all env vars in the container
+	envs := os.Environ()
+
+	// find all env vars match specified prefix
+	for _, env := range envs {
+		var envKey string
+		var envValue string
+		env_kv := strings.SplitN(env, "=", 2)
+		if len(env_kv) != 2 {
+			logrus.Fatalf("[FluentBit Init Process] Unrecognizable environment variables: %s\n", env)
+		}
+
+		envKey = string(env_kv[0])
+		envValue = string(env_kv[1])
+
+		matched_s3, _ := regexp.MatchString("aws_fluent_bit_s3_", envKey)
```

Comment:
Based on discussion, let's change this to `aws_fluent_bit_init_s3` and `aws_fluent_bit_init_file`

Reply:
And both `aws_fluent_bit_init_s3` and `aws_fluent_bit_init_S3` will be accepted as config files from S3.
I think it would be more helpful here not to use case sensitivity, customers may not be concerned about the way **s3** or **S3** is written